spec/unit/inventory_v4_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative '../spec_helper'
+
+include Hawkular::InventoryV4
+describe 'Inventory v4' do
+  let(:resource_hash) do
+    {
+      'id' => 'root',
+      'type' => {},
+      'children' => [
+        { 'id' => 'child-01', 'type' => { 'id' => 'type-01' }, 'children' => [] },
+        { 'id' => 'child-02', 'type' => { 'id' => 'type-02' },
+          'children' => [
+            { 'id' => 'child-03', 'type' => { 'id' => 'type-02' }, 'children' => [] },
+            { 'id' => 'child-04', 'type' => { 'id' => 'type-02' }, 'children' => [] },
+            { 'id' => 'child-05', 'type' => { 'id' => 'type-03' }, 'children' => [] }
+          ]
+        },
+        { 'id' => 'child-06', 'type' => { 'id' => 'type-01' }, 'children' => [] }
+      ]
+    }
+  end
+
+  describe 'children by type' do
```

Comment:
`describe '#children_by_type'`, since we're testing a method.

Reply:
@josejulio This is useful if you do already have the whole tree, but will you always have the whole tree in advance?  Would it make sense to have underlying support in the inventory UI to fetch resources by type?

Reply:
I can already fetch resources by type in the API. But in order to link them together, I have to group them by e.g. the feed_id. 

I'm trying to avoid searching by feed, instead fetching a whole tree and processing the tree. (discarding the tree after I finished the processing it)
That way I'm sure that the whole tree belongs to the same "feed".

Then extracting the required types from the tree (depending what is currently being processed).

Reply:
Don't hesitate to suggest API improvement on hawkular side. For instance, given our current indexing, I think we could perfectly add a "get child resource by type" query, given a type and a parent id.